packages/analytics/amplify_analytics_pinpoint/example/lib/main.dart

```diff
@@ -40,18 +40,19 @@ class _MyAppState extends State<MyApp> {
     if (!mounted) return;
 
     // Configure analytics plugin
-    final secureStorage = AmplifySecureStorage(
-      config: AmplifySecureStorageConfig(
-        scope: 'analytics',
-        // FIXME: In your app, make sure to remove this line and set up
-        /// Keychain Sharing in Xcode as described in the docs:
-        /// https://docs.amplify.aws/lib/project-setup/platform-setup/q/platform/flutter/#enable-keychain
-        // ignore: invalid_use_of_visible_for_testing_member
-        macOSOptions: MacOSSecureStorageOptions(useDataProtection: false),
-      ),
-    );
-    final authPlugin = AmplifyAuthCognito(credentialStorage: secureStorage);
-    final analyticsPlugin = AmplifyAnalyticsPinpoint();
+    storageFactory(AmplifyStorageScope scope) => AmplifySecureStorage(
+          config: AmplifySecureStorageConfig(
+            scope: scope.name,
+            // FIXME: In your app, make sure to remove this line and set up
+            /// Keychain Sharing in Xcode as described in the docs:
+            /// https://docs.amplify.aws/lib/project-setup/platform-setup/q/platform/flutter/#enable-keychain
+            // ignore: invalid_use_of_visible_for_testing_member
+            macOSOptions: MacOSSecureStorageOptions(useDataProtection: false),
+          ),
+        );
+    final authPlugin = AmplifyAuthCognito(secureStorageFactory: storageFactory);
+    final analyticsPlugin =
+        AmplifyAnalyticsPinpoint(secureStorageFactory: storageFactory);
```

Comment:
I'm wondering if we want to offer a new static method on AmplifySecureStrorage to make this API a little cleaner. We could also consider making the current constructor for AmplifySecureStorage internal since we don't really want to encourage use of in outside of our libs. Everyone should just be using the factory.

I need to put a little more thought into the naming and impl, but something like the below would work.

@dnys1, thoughts?

```dart
// main.dart
final macOSOptions = MacOSSecureStorageOptions(useDataProtection: false);
final authPlugin = AmplifyAuthCognito(
  secureStorageFactory: AmplifySecureStorage.fromFactory(
    macOSOptions: macOSOptions,
  ),
);
final analyticsPlugin = AmplifyAnalyticsPinpoint(
  secureStorageFactory: AmplifySecureStorage.fromFactory(
    macOSOptions: macOSOptions,
  ),
);
```

```dart
// amplify_secure_storage.vm.dart
class AmplifySecureStorage extends AmplifySecureStorageInterface {
  /// Generates a [AmplifySecureStorage] from a config.
  ///
  /// Should only be used within Amplify packages. See [fromFactory] for
  /// external use.
  @internal
  AmplifySecureStorage.fromConfig({
    required super.config,
  });

  /// Returns a factory for creating [AmplifySecureStorage] instances.
  static SecureStorageFactory fromFactory({
    WebSecureStorageOptions? webOptions,
    WindowsSecureStorageOptions? windowsOptions,
    LinuxSecureStorageOptions? linuxOptions,
    MacOSSecureStorageOptions? macOSOptions,
    IOSSecureStorageOptions? iOSOptions,
  }) {
    return (AmplifyStorageScope scope) {
      return AmplifySecureStorage.fromConfig(
        config: AmplifySecureStorageConfig(
          scope: scope.name,
          webOptions: webOptions,
          windowsOptions: windowsOptions,
          linuxOptions: linuxOptions,
          macOSOptions: macOSOptions,
          iOSOptions: iOSOptions,
        ),
      );
    };
  }
  ...
}

Reply:
I like this, it hides internal implementation details that customers should not be seeing. I think we will still need to expose the general constructor though because we have several unit tests that need to seed data on specific secure storage instances.

Reply:
Sounds good - should be called `factoryFrom` though or just `factory` so that it makes sense without arguments, e.g. `AmplifySecureStorage.factory()`

Reply:
Agreed. `factoryFrom` or just `factory` makes more sense.

Reply:
I think I prefer `factoryFrom` since you would never really call it without arguments - it's an optional parameter everywhere, so overriding it would only be to customize it in some way.